resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -536,17 +558,31 @@ private InetSocketAddress nextNameServerAddress() {
             Promise<AddressedEnvelope<? extends DnsResponse, InetSocketAddress>> promise) {
 
         return query0(checkNotNull(nameServerAddr, "nameServerAddr"),
-                      checkNotNull(question, "question"),
-                      checkNotNull(promise, "promise"));
+                checkNotNull(question, "question"), null, checkNotNull(promise, "promise"));
```

Comment:
I was thinking about that, but I was not sure what option is preferred in Netty. Btw, what about use of some @Nullable annotation? I don't see this kind of annotations in Netty...